files/image_config/memory_statistics/memory_statistics_service.py

```
@@ -0,0 +1,2134 @@
+        threads to finish their execution within a specified timeout.
+        Logs any issues encountered during the stopping process of the threads.
+        """
+        logger.log_info("Signaling threads to stop.")
+        self.stop_event.set()
+
+        if self.socket_listener_thread and self.socket_listener_thread.is_alive():
+            logger.log_info("Waiting for socket listener thread to stop...")
+            self.socket_listener_thread.join(timeout=5)
+            if self.socket_listener_thread.is_alive():
+                logger.log_warning("Socket listener thread did not terminate within the timeout period.")
+
+        if self.memory_collection_thread and self.memory_collection_thread.is_alive():
+            logger.log_info("Waiting for memory collection thread to stop...")
+            self.memory_collection_thread.join(timeout=5)
+            if self.memory_collection_thread.is_alive():
+                logger.log_warning("Memory collection thread did not terminate within the timeout period.")
+
+        logger.log_info("All threads stopped.")   
+
+    def cleanup(self):
+        """
+        Performs additional cleanup tasks before service shutdown.
+        This method handles the cleanup of old log files, removes the socket 
+        and PID files, and flushes all log handlers to ensure that all logs 
+        are properly written out. This is typically called during service 
+        termination to free up resources and maintain a clean state.
+        Logs the completion of the cleanup tasks and any errors encountered 
+        during the process.
+        """
+        self.cleanup_old_files()
+
+        try:
+            socket_address = self.config.get('DBUS_SOCKET_ADDRESS')
+            if socket_address and os.path.exists(socket_address):
+                os.unlink(socket_address)
+                logger.log_info(f"Removed socket file: {socket_address}")
+        except Exception as e:
```

>except Exception as [](http://example.com/codeflow?start=8&length=19)

still too general Exception, still swallow exception.

---

Thank you for the feedback.
I have replaced except Exception with more specific exceptions to ensure better debugging and logging.